src/main/services/correspondence-address/address-autolookup.ts

```diff
@@ -0,0 +1,73 @@
+import { UKAddress } from "@companieshouse/api-sdk-node/dist/services/postcode-lookup/types";
+import { Session } from "@companieshouse/node-session-handler";
+import { Request } from "express";
+import { ADDRESS_LIST, USER_DATA } from "../../common/__utils/constants";
+import { saveDataInSession } from "../../common/__utils/sessionHelper";
+import { ACSPData } from "../../model/ACSPData";
+import { Address } from "../../model/Address";
+import { getCountryFromKey } from "../../utils/web";
+
+export class CorrespondenceAddressAutoLookService {
+    saveCorrespondenceAddress (acspData :ACSPData, ukAddresses: UKAddress[], inputPremise: string): ACSPData {
+
+        let address = {
+            premise: "",
+            propertyDetails: "",
+            line1: "",
+            line2: "",
+            town: "",
+            country: "",
+            postcode: ""
+        };
+        for (const ukAddress of ukAddresses) {
+            if (ukAddress.premise === inputPremise) {
+                address = {
+                    propertyDetails: ukAddress.premise,
+                    premise: ukAddress.premise,
+                    line1: ukAddress.addressLine1,
+                    line2: ukAddress.addressLine2!,
+                    town: ukAddress.postTown,
+                    country: getCountryFromKey(ukAddress.country),
+                    postcode: ukAddress.postcode
+                };
+            }
```

Comment:
you can break out of the loop once the match is found

Reply:
There is no difference between address and correspondenceAddress. Only one can be used